net/error.go

```diff
@@ -0,0 +1,26 @@
+// Copyright 2019 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package net
+
+type ConnectionError struct {
+	// TODO: create status enums and move to metrics.go
+	Status  string
+	Message string
+	Cause   error
```

Comment:
*nit* In go standard library, they seem to name nested error(s) as `Err`: https://pkg.go.dev/io/fs#PathError.

Reply:
We can't change this without breaking some code. Also, I think `Cause` is a lot clearer.

In any case, this is existing code that is not changing and is out of scope of this refactoring.